src/components/UpdateVotes.jsx

```diff
@@ -0,0 +1,68 @@
+import PropTypes from "prop-types";
+import { useState } from "react";
+import {
+  IoHeartOutline,
+  IoHeart,
+  IoHeartDislike,
+  IoHeartDislikeOutline,
+} from "react-icons/io5";
+import { decreaseArticleVotes, increaseArticleVotes } from "../../api";
+import ErrorComponent from "./ErrorComponent";
+
+const UpdateVotes = ({ votes, setVotes, article_id }) => {
+  const [loveToggle, setLoveToggle] = useState(false);
+  const [hateToggle, setHateToggle] = useState(false);
+  const [error, setError] = useState(null);
+
+  const handleLoveClick = () => {
+    setLoveToggle(true);
+    increaseArticleVotes(article_id).catch((err) => {
+      setError(err);
+    });
+    setVotes((votes) => votes + 1);
```

Comment:
If you move this setting of state before the API call, then we'll have optimistic rendering. Same in the handleHateClick function 👍